src/OctoshiftCLI.Tests/AdoApiTests.cs

```diff
@@ -37,5 +42,216 @@ public async void Get_User_Id_Invalid_Response()
+
+            var mockClient = new Mock<AdoClient>(null, null);
+
+            mockClient.Setup(x => x.GetWithPagingAsync($"https://dev.azure.com/{adoOrg}/{teamProject1}/_apis/serviceendpoint/endpoints?api-version=6.0-preview.4").Result).Returns(JArray.Parse("[]"));
+            mockClient.Setup(x => x.GetWithPagingAsync($"https://dev.azure.com/{adoOrg}/{teamProject2}/_apis/serviceendpoint/endpoints?api-version=6.0-preview.4").Result).Returns(response);
+
+            using var sut = new AdoApi(mockClient.Object);
+            var result = await sut.GetGithubAppId(adoOrg, githubOrg, teamProjects);
+
+            Assert.Equal(appId, result);
+        }
+
+        [Fact]
+        public async void GetGithubAppIdTwoProjectsNoMatch()
+        {
+            var adoOrg = "foo-org";
+            var githubOrg = "foo-gh-org";
+            var teamProject1 = "foo-tp1";
+            var teamProject2 = "foo-tp2";
+            var teamProjects = new List<string>() { teamProject1, teamProject2 };
+            var appId = Guid.NewGuid().ToString();
+
+            var json = "[{type: 'GitHub', name: 'wrongOrg', id: '" + appId + "'}]";
+            var response = JArray.Parse(json);
+
+            var mockClient = new Mock<AdoClient>(null, null);
+
+            mockClient.Setup(x => x.GetWithPagingAsync($"https://dev.azure.com/{adoOrg}/{teamProject1}/_apis/serviceendpoint/endpoints?api-version=6.0-preview.4").Result).Returns(JArray.Parse("[]"));
+            mockClient.Setup(x => x.GetWithPagingAsync($"https://dev.azure.com/{adoOrg}/{teamProject2}/_apis/serviceendpoint/endpoints?api-version=6.0-preview.4").Result).Returns(response);
+
+            using var sut = new AdoApi(mockClient.Object);
+            var result = await sut.GetGithubAppId(adoOrg, githubOrg, teamProjects);
+
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public async void GetGithubHandle()
+        {
+            var endpoint = $"https://dev.azure.com/FOO-ORG/_apis/Contribution/HierarchyQuery?api-version=5.0-preview.1";
+            var payload = @"
+{
+    ""contributionIds"": [
+        ""ms.vss-work-web.github-user-data-provider""
+    ],
+    ""dataProviderContext"": {
+        ""properties"": {
+            ""accessToken"": ""FOO-TOKEN"",
+            ""sourcePage"": {
+                ""url"": ""https://dev.azure.com/FOO-ORG/FOO-TEAMPROJECT/_settings/boards-external-integration#"",
+                ""routeId"": ""ms.vss-admin-web.project-admin-hub-route"",
+                ""routeValues"": {
+                    ""project"": ""FOO-TEAMPROJECT"",
+                    ""adminPivot"": ""boards-external-integration"",
+                    ""controller"": ""ContributedPage"",
+                    ""action"": ""Execute"",
+                    ""serviceHost"": ""FOO-ORGID (FOO-ORG)""
+                }
+            }
+        }
+    }
+}";
+            var json = "{ \"dataProviders\": { \"ms.vss-work-web.github-user-data-provider\": { \"login\": 'FOO-LOGIN' } } }";
```

Comment:
nit: I would create a `const` for `FOO-LOGIN` (i.e `const string expectedLogin = "FOO-LOGIN"`)